app/models/release.rb

```diff
@@ -556,6 +576,22 @@ def approvals_blocking?
     !(approvals_overridden? || approvals_finished?)
   end
 
+  def copy_approvals_enabled?
+    copy_approvals? && release?
+  end
+
+  def copy_approval_restricted?
+    fetch_previous_finished_release.nil? || approval_items.present? || hotfix?
```

Comment:
**thought:** we need to rethink the usage of `approval_items.present?` because the code currently has 2 race conditions:

- If the job gets queued late, a user could add items in between and the copy would go through (the restriction would not come into play). You can repro this by adding a fake sleep.
- If the current user sees that they can copy, but someone else copies or adds items in another tab, the copy would again go through (create a duplicate set).

This is because the check is only on the button, not in the job or the controller layer. We should do a more robust check overall:

**controller**

Here we should do 2 separate checks upfront (eg. [before_action](https://github.com/tramlinehq/tramline/blob/main/app/controllers/release_metadata_controller.rb#L10)):

1. Previous release present + not hotfix = "cannot copy" err message
2. Approval items already present = "cannot copy because items already exist" err message

**view**

Disable button if any of the above two checks are true.

**job**

We should only check for the first check (don't check for approval items presence at all) – this means that we allow the copy to go through even if items are added in between, we won't stop this at all.